httpcache.go

```diff
@@ -498,6 +516,41 @@ func headerAllCommaSepValues(headers http.Header, name string) []string {
 	return vals
 }
 
+// cachedReadCloser is a wrapper around ReadCloser R that calls OnEOF
+// handler with a full copy of the content read from that ReadCloser
+// when EOF is read.
```

Comment:
s/when EOF is read/when EOF is reached/, for consistency with the doc string of `OnEOF`.